introspection/providers_tree.go

```diff
@@ -0,0 +1,58 @@
+package introspection
+
+// ProvidersTree is a struct which provider modules use to register their interest in providing
+// data to be recorded in a time slice of the running system.
+type ProvidersTree struct {
+	Runtime *RuntimeProviders
+	Conn    *ConnProviders
+	Stream  *StreamProviders
+}
+
+// synthetic used types to contextualise inputs/outputs and enforce correctness.
+type (
+	// ConnID represents a connection ID string
+	ConnID string
+	// StreamID represents a connection ID string
+	StreamID string
+)
+
+// Enums.
+type (
+	// StreamListQueryType is an Enum to represent the types of queries that can be made when looking up streams
+	StreamListQueryType int
```

Comment:
I think we can generalise this type and name it `QueryOutput`, with two concrete enum values: `QueryOutputList`, `QueryOutputFull`.

Reply:
I've come up with `QueryOutputTypeFull ` & `QueryOutputTypeIds ` which determine whether we want to resolve an output object completely or just fetch it's Ids.